apps/wallet-api-tools/src/components/GroupedMessage.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from "react";
+import { MessageGrouped } from "../types";
+import { getDate } from "../helpers";
+import { coolGlow, dracula } from "thememirror";
+import Message from "./Message";
+
+type Props = {
+  group: MessageGrouped["messages"];
+  setValue: (value: string) => void;
+  displayModal?: boolean;
+};
+
+const GroupedMessage = ({ group, setValue, displayModal }: Props) => {
+  const parsedSentValue = JSON.parse(group.sent.value) as {
+    id: string;
+    jsonrpc: string;
+    method: string;
+    [value: string]: string;
+  };
```

Comment:
We should probably parse in a `useMemo`

Reply:
I don't know if it's very useful as its not used into any child component